en/micro-integrator/docs/administer-and-observe/logs/configuring_log4j_properties.md

```diff
@@ -686,6 +686,73 @@ You can hide the 'Current Params' in the printed logs by passing the following s
 -Ddss.disable.current.params=true \
 ```
 
+## Using Custom Log Appenders
+
+Custom log appenders for Log4j2 can be used to store application logs in various environments/systems, such as cloud storages.
+
+But since WSO2 Micro Integrator works in an OSGi environment, such Log4j2 extensions cannot be used as they are. Therefore those extensions need to be modified to be compatible with WSO2 Micro Integrator. To modify an existing Log4j2 extension, follow the steps mentioned below.
+
+1. In the custom log appender, open the `pom.xml` file of the module which contains the `Log4j2Appender` class.
+
+2. Under the `build` section, add the `maven-compiler-plugin` and the `maven-bundle-plugin` as follows.
+
+    ```xml
+    <plugins>
+       ...
+       <plugin>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <executions>
+                <execution>
+                   <id>log4j-plugin-processor</id>
+                   <goals>
+                      <goal>compile</goal>
+                   </goals>
+                   <phase>process-classes</phase>
+                   <configuration>
+                      <proc>only</proc>
+                      <annotationProcessors>
+                            <annotationProcessor>
+                               org.apache.logging.log4j.core.config.plugins.processor.PluginProcessor
+                            </annotationProcessor>
+                      </annotationProcessors>
+                   </configuration>
+                </execution>
+          </executions>
+       </plugin>
+       <plugin>
+          <groupId>org.apache.felix</groupId>
+          <artifactId>maven-bundle-plugin</artifactId>
+          <extensions>true</extensions>
+          <configuration>
+                <instructions>
+                   <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                   <Bundle-Name>${project.artifactId}</Bundle-Name>
+                   <Fragment-Host>org.ops4j.pax.logging.pax-logging-log4j2</Fragment-Host>
+                   <Export-Package>
+                      <PACKAGE_CONTAINS_THE_APPENDER_CLASS>
+                   </Export-Package>
+                  <DynamicImport-Package>*</DynamicImport-Package>
+                   <Import-Package></Import-Package>
+                   <Include-Resource>${project.build.directory}/classes/</Include-Resource>
+                </instructions>
+          </configuration>
+       </plugin>
+       ...
+    ```
+
+3. Rebuild the related module and copy the built JAR file from the `target` directory to `<MI_HOME>/dropins` directory.
+
+4. Configure the custom appender as follows.
+
+    ```properties
```

Comment:
Shall we add something like **appenders = log4j2Custom** here? To let users know that we need to add the custom appender to the appenders list. WDYT?

Reply:
fixed